x/batching/keeper/data_result.go

```diff
@@ -50,60 +50,118 @@ func (k Keeper) GetDataResult(ctx context.Context, dataReqID string) (*types.Dat
 	return &dataResult, err
 }
 
+// GetLatestDataResult returns the latest data result given the associated
+// data request's ID.
+func (k Keeper) GetLatestDataResult(ctx context.Context, dataReqID string) (*types.DataResult, error) {
+	dataResult, err := k.getLatestDataResult(ctx, false, dataReqID)
+	if err != nil {
+		if errors.Is(err, collections.ErrNotFound) {
+			// Look among batched data requests.
+			dataResult, err := k.getLatestDataResult(ctx, true, dataReqID)
+			if err != nil {
+				if errors.Is(err, collections.ErrNotFound) {
+					return nil, nil
```

Comment:
Not returning an error here could cause a nil dereference error in the querier, right?

Reply:
Good catch. I guess adding a null check in the querier makes the most sense?